### PR TITLE
Memory leak fixes

### PR DESCRIPTION
**Pull request by MarcelHB**
## Description
Of #2188 I fixed 3 problems by us (visible in different reports), two things seem to be related to STL and are known issues to show false positives, one looks too complicated and asks for some better solution. I'll open a follow-up.

Still no idea about Windows metrics, will do something if I find out what's wrong, if anything at all.

## Checklist

- [x] Commit messages are descriptive and explain the rationale for changes
- [x] I used the same coding style as the surrounding code 
- [x] I have tested the proposed changes
- [x] I extended the documentation, if necessary
- [ ] The proposed change builds also on our build bots (check after submission)

### CODE REVIEW

File: gemrb/plugins/CREImporter/CREImporter.cpp [hunk 1]
```diff
@@ -428,6 +428,12 @@ static void GetKitSpell(const ResRef& tableRef, std::vector<SpellEntry*>& list)
 			index = FindSpell(spellRef, splList);
 			assert (index != TableMgr::npos);
 		}
+
+		// IWD2 listdomn gives us dups here
```
Comment (lynxlynxlynx) [inline gemrb/plugins/CREImporter/CREImporter.cpp:432]: spwi903 is the only one, at level 8 for Oghmatiques, at level 9 for CTE.
Anyway, wouldn't this way one of them not get the spell? I think the correct thing to do is to rerun an amended loop below if the entry already exists (overwrite values if they are 0). This is basically a data problem, the two adjacent lines should have been one.

Comment (MarcelHB) [reply gemrb/plugins/CREImporter/CREImporter.cpp:432]: The `SpellEntry` uses a vector for all this, so overwriting using the existing interface won't work. Will look at this tomorrow.

Comment (MarcelHB) [reply gemrb/plugins/CREImporter/CREImporter.cpp]: Minor code cleanup now.

**Comment by lynxlynxlynx** [comment]
Thanks!

**Pull request merged**